### PR TITLE
Delete/update tuit functions debugged and simplified

### PR DESCRIPTION
**Pull request by kachalmers**
Updated delete functions to use helper function in TuitService for deletion of related Tuit2Tags and updates/deletion of Tags when tuits are about to be deleted.

### CODE REVIEW

**Review by mcartervt — APPROVED** [review]
Code looks good.  Built and started on my machine.  Passed Postman tests.

Approved!

File: controllers/TuitController.ts [hunk 1]
```diff
@@ -214,59 +212,24 @@ export default class TuitController implements TuitControllerI {
...
-            .then(status => res.json(status))
+        // Turn tuit into array of one tuit to be given to helper function
+        const tuits2BeUpdated = [tuitToBeUpdated];
+
+        // Remove old tuit2tags and tags corresponding to given tuit
+        await TuitController.tuitService.deleteTuit2TagsAndUpdateTagsByTuits(tuits2BeUpdated);
+
+        // Update the tuit
+        await TuitController.tuitDao.updateTuit(req.params.tid, req.body)
+            .then(status => res.json(status));
```
Comment (mcartervt) [inline controllers/TuitController.ts:226]: Does it matter if we have code below the res?  I'm just not sure still if this is acting as a "return" type situation.

**Review by mcartervt — APPROVED** [review]
Works perfectly for me!  Build ran and started.  Code looks really clean with the helper function.  I tested via Postman with two tags, where one was a substring of the other.  It worked removing both and just one.

Approved!

**Review by klalmeida — APPROVED** [review]
Also builds and runs for me, everything looks good!

**Pull request merged**